charms/kfp-ui/tests/unit/test_operator.py

```diff
@@ -297,6 +298,70 @@ def test_install_with_all_inputs(harness, oci_resource_data):
     assert harness.charm.model.unit.status == ActiveStatus()
 
 
+def test_object_storage_secrets_encoded(harness, oci_resource_data):
+    harness.set_leader()
+    http_port = "1234"
+    model_name = "test_model"
+    harness.set_model_name(model_name)
+    harness.update_config({"http-port": http_port})
+
+    kfpui_relation_name = "kfp-ui"
+
+    harness.set_leader()
+
+    # Set up required relations
+    # Future: convert these to fixtures and share with the tests above
+    harness.add_oci_resource(**oci_resource_data)
+
+    # object storage relation
+    os_data = {
+        "_supported_versions": "- v1",
+        "data": yaml.dump(
+            {
+                "access-key": "access-key",
+                "namespace": "namespace",
+                "port": 1234,
+                "secret-key": "secret-key",
+                "secure": True,
+                "service": "service",
+            }
+        ),
+    }
+    os_rel_id = harness.add_relation("object-storage", "storage-provider")
+    harness.add_relation_unit(os_rel_id, "storage-provider/0")
+    harness.update_relation_data(os_rel_id, "storage-provider", os_data)
+
+    # kfp-api relation
+    kfpapi_data = {
+        "_supported_versions": "- v1",
+        "data": yaml.dump(
+            {
+                "service-name": "service-name",
+                "service-port": "1234",
+            }
+        ),
+    }
+    os_rel_id = harness.add_relation("kfp-api", "kfp-api-provider")
+    harness.add_relation_unit(os_rel_id, "kfp-api-provider/0")
+    harness.update_relation_data(os_rel_id, "kfp-api-provider", kfpapi_data)
+
+    relation_version_data = {"_supported_versions": "- v1"}
+
+    # example kfp-ui provider relation
+    kfpui_rel_id = harness.add_relation(kfpui_relation_name, f"{kfpui_relation_name}-subscriber")
+    harness.add_relation_unit(kfpui_rel_id, f"{kfpui_relation_name}-subscriber/0")
+    harness.update_relation_data(
+        kfpui_rel_id, f"{kfpui_relation_name}-subscriber", relation_version_data
+    )
+
+    harness.begin_with_initial_hooks()
+
+    pod_spec, _ = harness.get_pod_spec()
+    minio_secrets = pod_spec["kubernetesResources"]["secrets"][0]["data"]
+    assert b64decode(minio_secrets["MINIO_ACCESS_KEY"]).decode("utf-8") == "access-key"
+    assert b64decode(minio_secrets["MINIO_SECRET_KEY"]).decode("utf-8") == "secret-key"
```

Comment:
I'm not sure which is better, but is there a reason why we wouldn't put this test in the above `test_install_with_all_inputs` instead?  that is sort of an "end to end unit test" that does almost everything you need

Reply:
yeah it really is a style choice rather than there being right/wrong.  Where there's so much duplication I lean toward merging but ...